keras_contrib/layers/normalization.py

```diff
@@ -1,3 +1,236 @@
+            if your input tensor has shape (samples, channels, rows, cols),
+            set axis to 1 to normalize per feature map (channels axis).
+        momentum: momentum in the computation of the
+            exponential average of the mean and standard deviation
+            of the data, for feature-wise normalization.
+        r_max_value: Upper limit of the value of r_max.
+        d_max_value: Upper limit of the value of d_max.
+        t_delta: At each iteration, increment the value of t by t_delta.
+        weights: Initialization weights.
+            List of 2 Numpy arrays, with shapes:
+            `[(input_shape,), (input_shape,)]`
+            Note that the order of this list is [gamma, beta, mean, std]
+        beta_init: name of initialization function for shift parameter
+            (see [initializations](../initializations.md)), or alternatively,
+            Theano/TensorFlow function to use for weights initialization.
+            This parameter is only relevant if you don't pass a `weights` argument.
+        gamma_init: name of initialization function for scale parameter (see
+            [initializations](../initializations.md)), or alternatively,
+            Theano/TensorFlow function to use for weights initialization.
+            This parameter is only relevant if you don't pass a `weights` argument.
+        gamma_regularizer: instance of [WeightRegularizer](../regularizers.md)
+            (eg. L1 or L2 regularization), applied to the gamma vector.
+        beta_regularizer: instance of [WeightRegularizer](../regularizers.md),
+            applied to the beta vector.
+
+    # Input shape
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model.
+
+    # Output shape
+        Same shape as input.
+
+    # References
+        - [Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift](https://arxiv.org/abs/1502.03167)
+    """
+
+    def __init__(self, epsilon=1e-3, mode=0, axis=-1, momentum=0.99,
+                 r_max_val=3., d_max_val=5., t_delta=1., weights=None, beta_init='zero',
+                 gamma_init='one', gamma_regularizer=None, beta_regularizer=None,
+                 **kwargs):
+        self.supports_masking = True
+        self.beta_init = initializations.get(beta_init)
+        self.gamma_init = initializations.get(gamma_init)
+        self.epsilon = epsilon
+        self.mode = mode
+        self.axis = axis
+        self.momentum = momentum
+        self.gamma_regularizer = regularizers.get(gamma_regularizer)
+        self.beta_regularizer = regularizers.get(beta_regularizer)
+        self.initial_weights = weights
+        self.r_max_value = r_max_val
+        self.d_max_value = d_max_val
+        self.t_delta = t_delta
+        if self.mode == 0 or self.mode == 2:
```

Comment:
I always use batch norm mode 2 when doing tasks such as Image inpainting, super resolution or semantic segmentation. I find that not only does it perform better, sometimes with mode 0 it wont even train at all (as in this case, and a long time ago when I was doing super resolution with a GAN).

Also, I set r_max and d_max to 1e8 as a test (I dont want to try np.inf, it may be overkill). Training did not even budge until about the 17th epoch, and even then it rose only a little and then flatlined. So I think everything is working properly in BRN.